mzLib/Proteomics/Fragmentation/Product.cs

```diff
@@ -32,67 +41,52 @@ public struct Product
             SecondaryFragmentNumber = secondaryFragmentNumber;
         }
 
-        public string Annotation
-        {
-            get
-            {
-                StringBuilder sb = new StringBuilder();
-
-                if (SecondaryProductType == null)
-                {
-                    sb.Append(ProductType);
-
-                    // for "normal" fragments this is just the fragment number (e.g., the 3 in the b3 ion)
-                    // for diagnostic ions, it's the m/z assuming z=1
-                    // (e.g., a diagnostic ion with neutral mass 100 Da will be reported as the D101 fragment)
-                    sb.Append(FragmentNumber);
-                }
-                else
-                {
-                    //internal fragment ion, annotation used here: 10.1007/s13361-015-1078-1
-                    //example: yIb[18-36]
-                    sb.Append(ProductType + "I" + SecondaryProductType.Value + "[" + FragmentNumber + "-" + SecondaryFragmentNumber + "]");
-                }
-                if (NeutralLoss != 0)
-                {
-                    sb.Append("-");
-                    sb.Append(NeutralLoss.ToString("F2"));
-                }
-
-                return sb.ToString();
-            }
-        }
-
         /// <summary>
         /// Summarizes a Product into a string for debug purposes
         /// </summary>
         public override string ToString()
         {
             if (SecondaryProductType == null)
             {
-                return ProductType + "" + FragmentNumber + ";" + NeutralMass.ToString("F5") + "-" + string.Format("{0:0.##}", NeutralLoss);
+                return ProductType + "" + FragmentNumber + ";" + NeutralMass.ToString("F5") + "-" +
+                       string.Format("{0:0.##}", NeutralLoss);
             }
             else
             {
-                return ProductType + "I" + SecondaryProductType.Value + "[" + FragmentNumber + "-" + SecondaryFragmentNumber + "]" + ";" + NeutralMass.ToString("F5") + "-" + string.Format("{0:0.##}", NeutralLoss);
+                return ProductType + "I" + SecondaryProductType.Value + "[" + FragmentNumber + "-" +
+                       SecondaryFragmentNumber + "]" + ";" + NeutralMass.ToString("F5") + "-" +
+                       string.Format("{0:0.##}", NeutralLoss);
             }
         }
 
-        public override bool Equals(object obj)
+        public bool Equals(IProduct other)
         {
-            Product other = (Product)obj;
+            return NeutralMass.Equals(other.NeutralMass) && ProductType == other.ProductType &&
+                   NeutralLoss.Equals(other.NeutralLoss) && Terminus == other.Terminus &&
+                   FragmentNumber == other.FragmentNumber && Math.Abs(ResiduePosition - other.ResiduePosition) < 0.0001 &&
```

Comment:
ResiduePosition is an int. you can just do ResiduePosition == other.ResiduePosition

Reply:
fixed